File: tiffany-frontend/src/redux/features/location/store/models/response/location-response.ts
import { BasePagination } from "@/utils/common/pagination";

export interface ProvinceResponseModel {
  id: string;
  provinceCode: string;
  provinceEn: string;
  provinceKh: string;
}

export interface DistrictResponseModel {
  id: string;
  districtCode: string;
  districtEn: string;
  districtKh: string;
  provinceId?: string;
}

export interface CommuneResponseModel {
  id: string;
  communeCode: string;
  communeEn: string;
  communeKh: string;
  districtId?: string;
}

export interface VillageResponseModel {
  id: string;
  villageCode: string;
  villageEn: string;
  villageKh: string;
  communeId?: string;
}

export interface AllProvinceResponseModel extends BasePagination {
  content: ProvinceResponseModel[];
}

export interface AllDistrictResponseModel extends BasePagination {
  content: DistrictResponseModel[];
}

export interface AllCommuneResponseModel extends BasePagination {
  content: CommuneResponseModel[];
}

export interface AllVillageResponseModel extends BasePagination {
  content: VillageResponseModel[];
}

export interface LocationResponseModel {
  id: string;
  userId: string;
  label: string;
  village: string;
  commune: string;
  district: string;
  province: string;
  country: string;
  streetNumber: string;
  houseNumber: string;
  note: string;
  latitude: number;
  longitude: number;
  isDefault: boolean;
  fullAddress: string;
  hasCoordinates: boolean;
  createdAt: string;
  updatedAt: string;
  createdBy: string;
  updatedBy: string;
}

export interface AllLocationResponseModel extends BasePagination {
  content: LocationResponseModel[];
}

File: tiffany-backend/src/main/java/com/tiffany/features/location/dto/response/LocationResponse.java
package com.tiffany.features.location.dto.response;

import com.tiffany.shared.dto.BaseAuditResponse;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

@EqualsAndHashCode(callSuper = true)
@Data
public class LocationResponse extends BaseAuditResponse {
    private UUID userId;
    private String village;
    private String commune;
    private String district;
    private String province;
    private String streetNumber;
    private String houseNumber;
    private String note;
    private BigDecimal latitude;
    private BigDecimal longitude;
    private Boolean isDefault;
    private String fullAddress;
    private Boolean hasCoordinates;
    private List<LocationImageResponse> locationImages;
}